From 0fa9e7a6822e3cae2ede97d39d9a025f6a448c48 Mon Sep 17 00:00:00 2001
From: Rob Landers <landers.robert@gmail.com>
Date: Tue, 21 Feb 2017 11:12:28 -0500
Subject: [PATCH] wifi patch

This is based on a wifi patch for 4.9.x ... this brings back some critical elements

Signed-off-by: Rob Landers <landers.robert@gmail.com>

---
 drivers/net/wireless/marvell/mwifiex/cfg80211.c | 3 +++
 drivers/net/wireless/marvell/mwifiex/pcie.c     | 9 +++++++++
 2 files changed, 12 insertions(+)

diff --git a/drivers/net/wireless/marvell/mwifiex/cfg80211.c b/drivers/net/wireless/marvell/mwifiex/cfg80211.c
index 145cc4b..0b83101 100644
--- a/drivers/net/wireless/marvell/mwifiex/cfg80211.c
+++ b/drivers/net/wireless/marvell/mwifiex/cfg80211.c
@@ -418,6 +418,9 @@ mwifiex_cfg80211_set_power_mgmt(struct wiphy *wiphy,
 
 	ps_mode = enabled;
 
+	mwifiex_dbg(priv->adapter, ERROR, "jpw: hacking ps_mode to false\n");
+	ps_mode = 0;
+
 	return mwifiex_drv_set_power(priv, &ps_mode);
 }
 
diff --git a/drivers/net/wireless/marvell/mwifiex/pcie.c b/drivers/net/wireless/marvell/mwifiex/pcie.c
index 4db07da..642fb6e 100644
--- a/drivers/net/wireless/marvell/mwifiex/pcie.c
+++ b/drivers/net/wireless/marvell/mwifiex/pcie.c
@@ -1692,6 +1692,15 @@ static int mwifiex_pcie_process_cmd_complete(struct mwifiex_adapter *adapter)
 
 	pkt_len = *((__le16 *)skb->data);
 	rx_len = le16_to_cpu(pkt_len);
+
+	if (rx_len == 0) {
+	    mwifiex_dbg(adapter, ERROR,
+	        "0 byte cmdrsp\n");
+        mwifiex_map_pci_memory(adapter, skb, MWIFIEX_UPLD_SIZE,
+            PCI_DMA_FROMDEVICE);
+        return 0;
+	}
+
 	skb_put(skb, MWIFIEX_UPLD_SIZE - skb->len);
 	skb_trim(skb, rx_len);
 	skb_pull(skb, INTF_HEADER_LEN);
-- 
2.10.2

